cgi-bin/DW/Controller/Talk.pm

```diff
@@ -176,13 +180,30 @@ sub talkpost_do_handler {
     # For errors that aren't immediately fatal, collect them as we go and let
     # the user fix them all at once.
     my @errors;
+    my $form_auth_ok = 1;
+
+    # First, make sure we've got POST data and confirm that it's from a legit
+    # local form (and not some CSRF shenanigans). Usually that's a real POST,
+    # but sometimes it's a saved POST that we set aside while an OpenID
+    # commenter sorted out their identity.
+    if ( $r->did_post ) {
 
-    # If this is a GET (not POST), see if they're coming back from an OpenID
-    # identity server. If so, restore the POST hash we saved before they left.
-    if (   ( $GET->{'openid.mode'} eq 'id_res' || $GET->{'openid.mode'} eq 'cancel' )
+        # We only check form_auth for real POSTs; if they're coming back from
+        # OpenID, we already checked earlier.
+        unless ( LJ::check_form_auth( $POST->{lj_form_auth} ) ) {
+
+            # Form_auth failure is extra sketchy, so let's remember this and
+            # avoid calling some other functions later.
+            $form_auth_ok = 0;
+            push @errors, LJ::Lang::ml('/talkpost_do.tt.error.invalidform');
+        }
+    }
+    elsif (( $GET->{'openid.mode'} eq 'id_res' || $GET->{'openid.mode'} eq 'cancel' )
```

Comment:
I had submitted a patch to hush some warnings in this block, but since you just changed it, maybe go ahead and do it in this PR instead? https://github.com/dreamwidth/dw-free/pull/2743/commits/691c90694741cf73571496c9624ee830bba5296b

Reply:
Good call, thanks for the heads-up. I'll try and rig it so you show up as the commit author.